src/lib/support/TimeUtils.cpp

```diff
@@ -505,21 +520,43 @@ bool CalendarTimeToSecondsSinceEpoch(uint16_t year, uint8_t month, uint8_t dayOf
 void SecondsSinceEpochToCalendarTime(uint32_t secondsSinceEpoch, uint16_t & year, uint8_t & month, uint8_t & dayOfMonth,
                                      uint8_t & hour, uint8_t & minute, uint8_t & second)
 {
-    uint32_t daysSinceEpoch = secondsSinceEpoch / kSecondsPerDay;
-    uint32_t timeOfDay      = secondsSinceEpoch - (daysSinceEpoch * kSecondsPerDay);
-
-    // Note: This call to DaysSinceEpochToCalendarDate can't fail, because we
-    // can't overflow a uint16_t year with a 32-bit number of seconds.
-    static_assert(std::numeric_limits<decltype(secondsSinceEpoch)>::max() / (kDaysPerStandardYear * kSecondsPerDay) + 1 <=
-                      std::numeric_limits<std::remove_reference<decltype(year)>::type>::max(),
-                  "What happened to our year or day lengths?");
```

Comment:
Why did this comment and assert go away?  If this is no longer true, the `DaysSinceEpochToCalendarDate` call in the `uint64_t` version of `SecondsSinceEpochToCalendarTime` needs error-checking.  If it's still true, then we should have this bit in the `uint64_t` function, no?